src/Core/Context/CurrencyContext.php

```diff
@@ -26,34 +26,20 @@
 
 declare(strict_types=1);
 
-namespace PrestaShop\PrestaShop\Core\Model;
+namespace PrestaShop\PrestaShop\Core\Context;
 
 /**
- * @experimental Depends on ADR https://github.com/PrestaShop/ADR/pull/33
+ * This context service gives access to all contextual data related to currency.
  */
-interface EmployeeInterface
+class CurrencyContext
 {
-    public function getId(): int;
-
-    public function getProfileId(): int;
-
-    public function getLanguageId(): int;
-
-    public function getFirstName(): string;
-
-    public function getLastName(): string;
-
-    public function getEmail(): string;
-
-    public function getPassword(): string;
-
-    public function getImageUrl(): string;
-
-    public function getDefaultTabId(): int;
-
-    public function getDefaultShopId(): int;
-
-    public function getAssociatedShopIds(): array;
-
-    public function getAssociatedShopGroupIds(): array;
+    public function __construct(
+        private readonly Currency $currency
+    ) {
+    }
+
+    public function getCurrency(): Currency
+    {
+        return $this->currency;
+    }
```

Comment:
Apart from the shop, this layer goes flat, right?

Reply:
I'm not sure I follow you Do you mean it's overkill to have a dedicated `CurrencyContext` service/class that only contains one DTO? And the service could be the DTO itself?

If yes, I thought about it actually, but then I thought about two arguments to keep it this way:
- we may enrich this country context with something we forgot, or that we don't have/need yet If/when it happens we'd be happy to be able to easily add a getter/function without messing with the scope of the Currency DTO
- I think maybe we should favor consistency and make every new sub contexts behave the same, if Shop context is based on a triple class so should all the others

That said! The question is still valid and arguable, it's tempting to flatten the whole context into one class instead of ContextClass + DTO. But if we decide to do so then out of consistency maybe we should do the same for ShopContext and EmployeeContext I guess